packages/swapper/src/swappers/cow/getCowSwapTradeQuote/getCowSwapTradeQuote.ts

```diff
@@ -0,0 +1,137 @@
+        code: SwapErrorTypes.UNSUPPORTED_PAIR,
+        details: { buyAssetNamespace, sellAssetNamespace }
+      })
+    }
+
+    const { minimum, maximum } = await getCowSwapMinMax(deps, sellAsset, buyAsset)
+
+    const minQuoteSellAmount = bnOrZero(minimum).times(bn(10).exponentiatedBy(sellAsset.precision))
+
+    // making sure we do not have decimals for cowswap api (can happen at least from minQuoteSellAmount)
+    const normalizedSellAmount = normalizeIntegerAmount(
+      bnOrZero(sellAmount).eq(0) ? minQuoteSellAmount : sellAmount
+    )
+
+    /**
+     * /v1/quote
+     * params: {
+     * sellToken: contract address of token to sell
+     * buyToken: contractAddress of token to buy
+     * receiver: receiver address can be defaulted to "0x0000000000000000000000000000000000000000"
+     * validTo: time duration during which quote is valid (eg : 1654851610 as timestamp)
+     * appData: appData for the CowSwap quote that can be used later, can be defaulted to "0x0000000000000000000000000000000000000000000000000000000000000000"
+     * partiallyFillable: false
+     * from: sender address can be defaulted to "0x0000000000000000000000000000000000000000"
+     * kind: "sell" or "buy"
+     * sellAmountBeforeFee / buyAmountAfterFee: amount in base unit
+     * }
+     */
+    const quoteResponse: AxiosResponse<CowSwapQuoteResponse> =
+      await cowService.post<CowSwapQuoteResponse>(`${deps.apiUrl}/v1/quote/`, {
+        sellToken: sellAssetErc20Address,
+        buyToken: buyAssetErc20Address,
+        receiver: DEFAULT_ADDRESS,
+        validTo: DEFAULT_VALIDTO_TIMESTAMP,
+        appData: DEFAULT_APP_DATA,
+        partiallyFillable: false,
+        from: DEFAULT_ADDRESS,
+        kind: ORDER_KIND_SELL,
+        sellAmountBeforeFee: normalizedSellAmount
+      })
+
+    const { data } = quoteResponse
+    const quote = data.quote
+
+    const rate = bn(quote.buyAmount)
+      .div(quote.sellAmount)
+      .times(bn(10).exponentiatedBy(sellAsset.precision - buyAsset.precision))
+      .toString()
+
+    const receiveAddress = await adapter.getAddress({ wallet })
+    const feeDataOptions = await adapter.getFeeData({
+      to: COW_SWAP_VAULT_RELAYER_ADDRESS,
+      value: normalizedSellAmount,
+      chainSpecific: { from: receiveAddress, contractAddress: sellAssetErc20Address },
+      sendMax: true
+    })
+    const feeData = feeDataOptions['fast']
+
+    const usdRateSellAsset = await getUsdRate(deps, sellAsset)
+    const feeUsd = bnOrZero(quote.feeAmount)
+      .div(bn(10).exponentiatedBy(sellAsset.precision))
+      .multipliedBy(bnOrZero(usdRateSellAsset))
+      .toString()
+
+    return {
+      rate,
+      minimum,
+      maximum,
+      feeData: {
+        fee: feeUsd,
+        chainSpecific: {
+          estimatedGas: feeData.chainSpecific.gasLimit,
+          gasPrice: feeData.chainSpecific.gasPrice,
+          approvalFee: bnOrZero(APPROVAL_GAS_LIMIT)
```

Comment:
Zrx only populates approvalFee if selling erc20 asset. would be good to follow the same pattern. no need to approval fee on trading from eth.

OR is cowswap only erc20? if thats case ignore my comment